### DISCUSSION TITLE
Loki: Add step editor

### DISCUSSION THREAD

**Discussion by ivanahuckova** [grafana/grafana#39022] [Data Sources]
In https://github.com/grafana/grafana/pull/36710 we have added possibility to specify desired resolution for Loki queries. We have considered adding step, but decided to use just resolution as Grafana should calculate the best step (1 data point for each pixel). Moreover, users are able to modify step trough `min interval` in dashboard query options. 

Based on some discussions, adding step would be helpful for some scenarios such as:

- I regularly force the step to match what I put in the range [1m] value to force the behavior of query. For example I want to force the buckets to be 1h or 1d, I would do [1h] and force the step to 1h.
- Alternating step for performance reasons, when you want to set higher step.
- In dashboard, for fine tuning the queries, when you want to receive only specific amount of results. But in this case, query option's min step can be used.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!

**Comment by ivanahuckova** [comment]
This was fixed and Loki now has step editor.
